codec-base/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -162,6 +164,8 @@ public ByteBuf cumulate(ByteBufAllocator alloc, ByteBuf cumulation, ByteBuf in)
     private static final byte STATE_CALLING_CHILD_DECODE = 1;
     private static final byte STATE_HANDLER_REMOVED_PENDING = 2;
 
+    // Used to guard the inputs for reentrant channelRead calls
+    private final Deque<Object> inputMessages = new ArrayDeque<>();
```

Comment:
should we better instance it lazy as this should not happen in most cases when using "real transports" ?

Reply:
agreed